### ISSUE TITLE
Why my loss < 0

### ISSUE THREAD

**Issue by WenxiongLiao** [MIC-DKFZ/nnUNet#662]
According to the guidelines of nnUNet, we train nnUNet on our own data set. But we found that **loss < 0** during training

We have repeatedly checked our data processing process, but still **loss<0**

**I am curious if this is normal? Or what is wrong?**

2021-06-17 13:00:20.939190: 
epoch:  0 
2021-06-17 13:07:16.926911: train loss : **-0.0682** 
2021-06-17 13:09:05.590441: validation loss: **0.0401** 
2021-06-17 13:09:05.591222: Average global foreground Dice: [0.0] 
2021-06-17 13:09:05.591390: (interpret this as an estimate for the Dice of the different classes. This is not exact.) 
2021-06-17 13:09:06.004262: lr: 0.009991 
2021-06-17 13:09:06.004472: This epoch took 525.065161 s
 
2021-06-17 13:09:06.004544: 
epoch:  1 
2021-06-17 13:18:47.696836: train loss : **-0.0912** 
2021-06-17 13:19:37.492402: validation loss: **-0.1148** 
2021-06-17 13:19:37.493006: Average global foreground Dice: [0.46969982247565767] 
2021-06-17 13:19:37.493114: (interpret this as an estimate for the Dice of the different classes. This is not exact.) 
2021-06-17 13:19:38.032607: lr: 0.009982 
2021-06-17 13:19:38.099739: saving checkpoint... 
2021-06-17 13:19:40.243560: done, saving took 2.21 seconds 
2021-06-17 13:19:40.391529: This epoch took 634.386921 s
 
2021-06-17 13:19:40.391732: 
epoch:  2 
2021-06-17 13:26:18.143355: train loss : **-0.1529** 
2021-06-17 13:27:06.083521: validation loss: **-0.1084** 
2021-06-17 13:27:06.083891: Average global foreground Dice: [0.25871622354634083] 
2021-06-17 13:27:06.083954: (interpret this as an estimate for the Dice of the different classes. This is not exact.) 
2021-06-17 13:27:06.561915: lr: 0.009973 
2021-06-17 13:27:06.615673: saving checkpoint... 
2021-06-17 13:27:07.604384: done, saving took 1.04 seconds 
2021-06-17 13:27:07.619365: This epoch took 447.227525 s
 
2021-06-17 13:27:07.619498: 
epoch:  3 
2021-06-17 13:33:47.752293: train loss : **-0.3014** 
2021-06-17 13:34:39.153103: validation loss: **-0.1449** 
2021-06-17 13:34:39.153476: Average global foreground Dice: [0.2337131222616985] 
2021-06-17 13:34:39.153544: (interpret this as an estimate for the Dice of the different classes. This is not exact.) 
2021-06-17 13:34:39.625314: lr: 0.009964 
2021-06-17 13:34:39.684065: saving checkpoint... 
2021-06-17 13:34:41.511213: done, saving took 1.89 seconds 
2021-06-17 13:34:41.525673: This epoch took 453.906089 s
 
2021-06-17 13:34:41.525804: 
epoch:  4 
2021-06-17 13:41:22.538193: train loss : **-0.3237** 
2021-06-17 13:43:13.985007: validation loss: **-0.1153** 
2021-06-17 13:43:13.985349: Average global foreground Dice: [0.33753968097482107] 
2021-06-17 13:43:13.985409: (interpret this as an estimate for the Dice of the different classes. This is not exact.) 
2021-06-17 13:43:14.445410: lr: 0.009955 
2021-06-17 13:43:14.504940: saving checkpoint... 
2021-06-17 13:43:16.297445: done, saving took 1.85 seconds 
2021-06-17 13:43:16.312203: This epoch took 514.786317 s
 
2021-06-17 13:43:16.312336: 
epoch:  5 
2021-06-17 13:49:54.877274: train loss : **-0.3480** 
2021-06-17 13:50:20.949013: validation loss: **-0.2022** 
2021-06-17 13:50:20.949445: Average global foreground Dice: [0.41406581390745334] 
2021-06-17 13:50:20.949539: (interpret this as an estimate for the Dice of the different classes. This is not exact.) 
2021-06-17 13:50:21.436844: lr: 0.009946 
2021-06-17 13:50:21.489840: saving checkpoint... 
2021-06-17 13:50:22.776952: done, saving took 1.34 seconds 
2021-06-17 13:50:22.787357: This epoch took 426.474949 s

**Comment by elpequeno** [comment]
The loss is supposed to be negative. The best loss is -1 due to it being the sum of CE (best is 0) and Dice (best is -1) loss. In a nutshell, the dice loss can be implemented as (1-dice) (here the best loss would be 0) or as (-dice) (here the best loss is -1). nnUnet uses the latter.

**Comment by FabianIsensee** [comment]
Yeah there is absolutely no reason to use the 1- formulation except that people are expecting the best loss value to be 0 :upside_down_face:
